### ISSUE TITLE
no-multiple-empty-lines encounters a TypeError

### ISSUE THREAD

**Issue by nclsndr** [eslint/eslint#8646]
**Environment**

* **ESLint Version:** 3.19.0
* **Node Version:** 4.2.0
* **npm Version:** 6.10.2

**Parser:** Babel-ESLint (same problem w/ default)

**Please show your full configuration:**

```
{
  "extends": "airbnb",
  "ecmaVersion": 7,
  "sourceType": "module",
  "ecmaFeatures": {
    "jsx": true
  },
  "env": {
    "browser": true
  },
  "plugins": [
  ],
  "parser": "babel-eslint",
  "rules": {
    "arrow-body-style": [0, "as-needed"],
    "arrow-parens": [2, "as-needed"],
    "comma-dangle": [2, "never"],
    "prefer-const": [2, {
      "destructuring": "all",
      "ignoreReadBeforeAssign": false
    }],
    "global-require": [0],
    "no-useless-escape": [0],
    "no-unused-vars": [0, { "vars": "local", "args": "after-used" }],
    "semi": [2, "never"],
    "prefer-arrow-callback": 2,
    "no-confusing-arrow": 0,
    "no-multiple-empty-lines": [2, { "max": 2 }],
    "no-return-assign": 1,
    "no-underscore-dangle": 0,
    "react/prop-types": [2, { "ignore": ["children", "dispatch"] }],
    "react/jsx-first-prop-new-line": [1, "multiline"],
    "react/jsx-no-bind": [1, { "allowBind": true }],
    "react/no-array-index-key": 1,
    "import/no-named-as-default": 0
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
const makeUrl = path => `${API_URL}${path}`



const Api = {
  any: 'hello'
}
```

**What did you expect to happen?**

the `no-multiple-empty-lines: [2, { "max": 2 }]` should display an eslint error

**What actually happened? Please include the actual, raw output from ESLint.**

I got a parse error like:
```
TypeError: sourceCode.getIndexFromLoc is not a function

TypeError: sourceCode.getIndexFromLoc is not a function
at Object.fix (/Users/nclsndr/Code/any/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:115:52)
at RuleContext.report (/Users/nclsndr/Code/any/node_modules/eslint/lib/rule-context.js:127:34)
at Program:exit.allLines.reduce.concat.reduce (/Users/nclsndr/Code/any/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:108:37)
at Array.reduce (native)
at EventEmitter.Program:exit (/Users/nclsndr/Code/any/node_modules/eslint/lib/rules/no-multiple-empty-lines.js:93:22)
at emitOne (events.js:101:20)
at EventEmitter.emit (events.js:188:7)
at NodeEventGenerator.leaveNode (/Users/nclsndr/Code/any/node_modules/eslint/lib/util/node-event-generator.js:48:22)
at CodePathAnalyzer.leaveNode (/Users/nclsndr/Code/any/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:627:23)
at CommentEventGenerator.leaveNode (/Users/nclsndr/Code/any/node_modules/eslint/lib/util/comment-event-generator.js:111:23)
at Controller.leave (/Users/nclsndr/Code/any/node_modules/eslint/lib/eslint.js:931:36)
Process finished with exit code -1
```
Thank you for your help!

**Comment by gyandeeps** [comment]
is this still an error if you try without `babel-eslint` parser?

**Comment by nclsndr** [comment]
@gyandeeps yep!

**Comment by ilyavolodin** [comment]
@nclsndr In that case, this is probably an issue with `babel-eslint` and should be opened there.

**Comment by mysticatea** [comment]
I'm not sure "`sourceCode.getIndexFromLoc` is not a function" is related to `babel-eslint` because it's a method of `SourceCode` object...

**Comment by not-an-aardvark** [comment]
Based on your stack trace:

```
at EventEmitter.emit (events.js:188:7)
at NodeEventGenerator.leaveNode (/Users/nclsndr/Code/any/node_modules/eslint/lib/util/node-event-generator.js:48:22)
```

It looks like you're using a version of ESLint before 3.18.0, because the stack trace for versions after 3.18.0 would include `applySelectors`.

However, I'm not sure why `sourceCode.getIndexFromLoc` is getting called if it doesn't exist in the current version.

**Comment by nclsndr** [comment]
@ilovejary I've also tried with `default` parser and it gave the same error
@not-an-aardvark I had the same idea — I've checked and: 
`$ eslint -v` gives `3.19.0`
`$ ./node_modules/.bin/eslint -v`gives `3.19.0`

I can't figure out where it comes from!

**Comment by not-an-aardvark** [comment]
What command are you using to run eslint? Does the problem still occur if you uninstall and reinstall it?

**Comment by not-an-aardvark** [comment]
Closing because no response from the author. If you're able to reproduce this issue again, feel free to create another issue.
